packages/gatsby/cache-dir/app.js

```diff
@@ -24,14 +35,21 @@ import "./blank.css"
 // Enable fast-refresh for virtual sync-requires, gatsby-browser & navigation
 // To ensure that our <Root /> component can hot reload in case anything below doesn't
 // satisfy fast-refresh constraints
-module.hot.accept(
-  [`$virtual/async-requires`, `./api-runner-browser`, `./navigation`],
-  () => {
-    // asyncRequires should be automatically updated here (due to ESM import and webpack HMR spec),
-    // but loader doesn't know that and needs to be manually nudged
-    loader.updateAsyncRequires(asyncRequires)
-  }
-)
+if (module.hot) {
+  module.hot.accept(
+    [
+      `$virtual/async-requires`,
+      `./.cache/_this_is_virtual_fs_path_/$virtual/async-requires.js`,
```

Comment:
`module.hot.accept` seems to not handle alias (`$virtual` is an alias) and require actual module path